### PR TITLE
mat64: export dense matrix IsZero methods

### PR DESCRIPTION
**Pull request by kortschak**
Blocked on #394.

@btracey Please take a look.

Not wildly happy with the comment wording.

### CODE REVIEW

File: mat64/dense.go [hunk 1]
```diff
@@ -136,7 +136,22 @@ func (m *Dense) isolatedWorkspace(a Matrix) (w *Dense, restore func()) {
...
+//
+// See the Reseter interface for more information.
+func (m *Dense) Reset() {
+	// Row, Cols and Stride must be zeroed in unison.
+	m.mat.Rows, m.mat.Cols, m.mat.Stride = 0, 0, 0
+	m.capRows, m.capCols = 0, 0
+	m.mat.Data = m.mat.Data[:0]
+}
+
+// IsZero returns whether the receiver is a zero value. A zero value may be
```
Comment (btracey) [inline mat64/dense.go]: Maybe: `IsZero returns whether the receiver is zero-sized. Zero-sized matrices can be the receiver for size-restricted operations. Dense matrices can be zeroed using Reset()`

**Comment by btracey** [comment]
Just saw your comment in stat. You're right, we should just be able to check the dimensions, so maybe it's better to keep this private.

**Comment by kortschak** [comment]
That works at the moment, but we could change it unless we document that that is the definition.

**Comment by kortschak** [comment]
Pending in gonum/gonum.

**Pull request closed without merging**